vyper/codegen/abi.py

```diff
@@ -373,12 +373,12 @@ def abi_encode(dst, lll_node, pos=None, bufsz=None, returns=False):
 
         elif isinstance(o.typ, BaseType):
             d = LLLnode(dst_loc, typ=o.typ, location='memory')
-            lll_ret.append(make_setter(d, o, "memory", pos))
+            lll_ret.append(make_setter(d, o, location=d.location, pos=pos))
         elif isinstance(o.typ, ByteArrayLike):
             d = LLLnode.from_list(dst_loc, typ=o.typ, location='memory')
             lll_ret.append(
                     ['seq',
-                        make_setter(d, o, o.location, pos=pos),
+                        make_setter(d, o, location=o.location,  pos=pos),
```

Comment:
I think I originally imagined that location would always be memory but I guess we should just fix this to always be `dst.location`, and also set `d`'s location to `dst.location`.